### PR TITLE
completed

### PR DESCRIPTION
**Pull request by 1102-Briski-Sean**
# Submitting Week 4 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [X] Behavioral Questions
- [X] Career Readiness
- [X] Coding Challenges
- [X] Interview Questions
- [X] Ruby Challenges

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback

If you would like specific feedback please leave details here:

### CODE REVIEW

File: behavioral-questions.md [hunk 1]
```diff
@@ -4,16 +4,16 @@ Answer the following questions. Use the Nugget + STAR method for each answer.
...
-- Situation:
-- Task:
-- Action:
-- Result:
\ No newline at end of file
+- Nugget: In the Navy, I spent two years in Norfolk, VA as an assistant weather forecaster.
+- Situation: I was fortunate to have been surrounded by many wonderful people, from the officers, to the civilians, to my fellow peers.
+- Task: As an assistant, I believed it was my duty, first and foremost, to learn as much as I could before I left for C-School. I needed to become competent and dependable. But I also needed to be a good teammate.
+- Action: And part of being a good teammate is allowing others to have the opportunity to grow and show their own value. Sometimes that meant sharing what I had learned and training others. Sometimes that meant putting those under me in roles they might not want, but would challenge them and help the team. Sometimes that meant taking on a role I may not want, but would allow other teammates to focus on pursuing other roles and career milestones. And sometimes, it just meant trusting my teammates and staying out of their way.
+- Result: I do not know how beneficial any of this actually was. I was just trying to do the best I could.
```
Comment (elysemontano) [inline behavioral-questions.md:19]: For this answer, I think focusing on one specific scenario that you may have a measurable result could make this answer stand out.  Also, consider modifying the verbiage "those under me" to "those I supervise" while focusing on the positive outcome of how you gave them a chance to grow.

File: code-challenges.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+// This file was created to test the legitimacy of the code created for the first problem of the coding challenge. I initially passed the test, but I couldn't tell if my code was actually shuffling the values in the array. After running the code in the terminal through this file, it became apparent that my code was only removing the first value of the array.
+// After much trial and error through this file, I was able to create a code that satisfied the conditions of the prompt.
```
Comment (elysemontano) [inline code-challenges.js:2]: You can also console log inside a function inside your test file.  This still allows you to troubleshoot and check outcomes without creating additional files.

File: code-challenges.test.js [hunk 3]
```diff
@@ -20,9 +20,53 @@ const colors1 = ["purple", "blue", "green", "yellow", "pink"]
...
+        array.pop()
+
+        // Transfer the values from the temporary array back into the modified array
+        for (let i = temporaryArray.length - 1; i >= 0; i = i - 1) {
+            array.push(temporaryArray[i])
+            temporaryArray.pop()
+        }
+    }
+    return shuffledArray
+}
```
Comment (elysemontano) [inline code-challenges.test.js:69]: Logic looks great on this problem!  If you are looking for a challenge, try refactoring this down to one for loop or even using a higher-order function.

File: ruby-challenges.rb [hunk 4]
```diff
@@ -33,4 +59,20 @@
...
 
-# Pseudo code:
\ No newline at end of file
+# Pseudo code:
+# Input: string
+# Output: string interpolation
+# Create conditional statement to see if word is the same spelled forward or backwards
+# Use .upcase to compare uppercase values
+
+def palindrome? string
```
Comment (elysemontano) [inline ruby-challenges.rb:68]: Nice work in this section!

**Review by elysemontano — APPROVED** [review]
Nice work Sean!  This assessment displays a competent level of skill acquisition.  I want to encourage you to consider refactoring your code once you have your tests passing.  There are oftentimes many different approaches that can achieve the same result, and it will only make you a better developer by exploring a variety of solutions.  I also left a few comments to take a look at.  Keep up the great work!

**Comment by 1102-Briski-Sean** [comment]
Thank you for your critique. I will keep your advice and suggestions in
mind.

With Gratitude,
Sean Briski

**Pull request merged**